src/aks-preview/azext_aks_preview/_validators.py

```diff
@@ -206,8 +208,17 @@ def validate_sku_tier(namespace):
     if namespace.tier is not None:
         if namespace.tier == '':
             return
-        if namespace.tier.lower() not in (CONST_MANAGED_CLUSTER_SKU_TIER_FREE, CONST_MANAGED_CLUSTER_SKU_TIER_STANDARD):
-            raise InvalidArgumentValueError("--tier can only be free or standard")
+        if namespace.tier.lower() not in (CONST_MANAGED_CLUSTER_SKU_TIER_FREE, CONST_MANAGED_CLUSTER_SKU_TIER_STANDARD, CONST_MANAGED_CLUSTER_SKU_TIER_PREMIUM):
+            raise InvalidArgumentValueError("--tier can only be free, standard, or premium")
+
+
+def validate_k8s_support_plan(namespace):
+    if namespace.k8s_support_plan is not None:
+        if namespace.k8s_support_plan == '':
+            return
+        if namespace.k8s_support_plan.lower() not in (
+                KubernetesSupportPlan.KUBERNETES_OFFICIAL.lower(), KubernetesSupportPlan.AKS_LONG_TERM_SUPPORT.lower()):
+            raise InvalidArgumentValueError("--k8s-support-plan can only be KubernetesOfficial or AKSLongTermSupport")
```

Comment:
Since the param is defined with `arg_type=get_enum_type(k8s_support_plans)` in _params.py, it is verfied validity automatically.

Reply:
@yanzhudd  Comment addressed. Thank you!